docs/platform-integration/storage/secure-storage.md

```diff
@@ -0,0 +1,151 @@
+
+-----
+<!-- markdownlint-enable MD025 -->
+
+## Use secure storage
+
+The following code examples demonstrate how to use secure storage.
+
+> [!TIP]
+> It's possible that an exception is thrown when calling `GetAsync` or `SetAsync`. This can be caused by a device not supporting secure storage, encryption keys changing, or corruption of data. it's best to handle this by removing and adding the setting back if possible.
+
+### Write a value
+
+To save a value for a given _key_ in secure storage:
+
+:::code language="csharp" source="../snippets/shared_1/Storage.cs" id="secstorage_set":::
+
+### Read a value
+
+To retrieve a value from secure storage:
+
+:::code language="csharp" source="../snippets/shared_1/Storage.cs" id="secstorage_get":::
+
+> [!TIP]
+> If there isn't a value associated with the key, `GetAsync` returns `null`.
+
+### Remove a value
+
+To remove a specific value, remove the key:
+
+:::code language="csharp" source="../snippets/shared_1/Storage.cs" id="secstorage_remove":::
+
+To remove all values, use the `RemoveAll` method:
+
+:::code language="csharp" source="../snippets/shared_1/Storage.cs" id="secstorage_remove_all":::
+
+## Platform differences
+
+This section describes the platform-specific differences with the secure storage API.
+
+<!-- markdownlint-disable MD025 -->
+<!-- markdownlint-disable MD024 -->
+# [Android](#tab/android)
+
+The [Android KeyStore](https://developer.android.com/training/articles/keystore.html) is used to store the cipher key used to encrypt the value before it's saved into a [Shared Preferences](https://developer.android.com/training/data-storage/shared-preferences.html) with a filename of _[YOUR-APP-PACKAGE-ID].xamarinessentials_. The value-key used in the shared preferences file is an **MD5** hash of the key passed into the `SecureStorage` APIs.
+
+- **API Level 23 and Higher**
+
+  On newer API levels, an **AES** key is obtained from the Android KeyStore and used with an **AES/GCM/NoPadding** cipher to encrypt the value before it's stored in the shared preferences file.
+
+- **API Level 22 and Lower**
+
+  On older API levels, the Android KeyStore only supports storing **RSA** keys, which is used with an **RSA/ECB/PKCS1Padding** cipher to encrypt an **AES** key, which is randomly generated at runtime. The **AES** key is stored in the shared preferences file under the key _SecureStorageKey_, but only if one wasn't already generated.
+
+**SecureStorage** uses the [Preferences](preferences.md) API and follows the same data persistence outlined in the [Preferences](preferences.md#persistence) documentation. If a device upgrades from API level 22 or lower, to API level 23 and higher, API Level 22 encryption is used unless the app is uninstalled or **RemoveAll** is called.
+
+# [iOS](#tab/ios)
+
+[KeyChain](xref:Security.SecKeyChain) is used to store values securely on iOS devices. The `SecRecord` used to store the value has a `Service` value set to _[YOUR-APP-BUNDLE-ID].xamarinessentials_.
```

Comment:
Same comment about filename.

Reply:
It is `$"{ApplicationModel.AppInfo.Current.PackageName}.microsoft.maui.essentials.{feature}"`